test/distributed/test_data_parallel.py

```diff
@@ -855,5 +769,102 @@ def forward(self, inp):
             model(input)
 
 
+class TestDataParallelDeviceType(TestCase):
```

Comment:
@cbalioglu Could you make sure that these 5 tests are still okay? I haven't changed their function, but rather just moved them to be a different base class so we could use @dtypes instead of using the outdated repeat_test_for_types. I also replaced the "cuda" in these tests with the device variable.